css/css-values/lh-rlh-on-root-001.html

```diff
@@ -0,0 +1,45 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Values and Units Test: using lh and rlh units on the root element</title>
+<link rel="author" title="Florian Rivoal" href="https://florian.rivoal.net/">
+<link rel="help" href="https://drafts.csswg.org/css-values-4/#font-relative-lengths">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<script>
+  function get_root_font_size() {
+    return window.getComputedStyle(window.document.documentElement).font_size;
+  }
+  function get_root_line_height() {
+    return window.getComputedStyle(window.document.documentElement).lineHeight;
+  }
+
+  window.document.documentElement.style="font-size: initial; line-height:initial;";
+  initial_f_s = get_root_font_size();
+  initial_l_h = get_root_line_height()
+
+  test(function() {
+    window.document.documentElement.style="font-size: 42px; line-height: 1lh;";
+    l_h = get_root_line_height();
+    assert_equals( l_h, initial_l_h, "the lh unit on the root element's line-height property uses font metrics corresponding to the initial values of the font or line-height properties");
+  }, "lh in line-height on root");
+
+  test(function() {
+    window.document.documentElement.style="font-size: 42px; line-height: 1rlh;";
+    l_h = get_root_line_height();
+    assert_equals( l_h, initial_l_h, "the rlh unit on the root element's line-height property uses font metrics corresponding to the initial values of the font or line-height properties");
+  }, "rlh in line-height on root");
+
+  test(function() {
+    window.document.documentElement.style="font-size: 1lh; line-height: 42px;";
+    f_s = get_root_font_size();
+    assert_equals( f_s, initial_f_s, "the lh unit on the root element's font-size property uses font metrics corresponding to the initial values of the font or line-height properties");
+  }, "lh in font-size on root");
+
+  test(function() {
+    window.document.documentElement.style="font-size: 1rlh; line-height: 42px;";
+    f_s = get_root_font_size();
+    assert_equals( f_s, initial_f_s, "the rlh unit on the root element's font-size property uses font metrics corresponding to the initial values of the font or line-height properties");
+
+  }, "rlh in font-size on root");
+</script>
```

Comment:
Shouldn't there be a test for what happens if you specify 2lh or 2rlh?

Reply:
No sure. In general, having a test checking that the unit can be use with numbers other than 1 sounds sane, but I am not sure it is needed as part of the unit cyclic dependency resolution problem, which is what I am testing here. On top of that, I am a little worried here that I'd easily run into rounding errors on this test if I started doing multiplications, especially for `rlh` where I have no way to for the initial value to be a round number.

As for testing that the unit in general works and makes sense with numbers other than one, I'm planning to write another series of tests about these units checking that they are sized correctly, now that we have resolved on https://github.com/w3c/csswg-drafts/issues/1253. Would that be enough, or do you think there is a need for this within the cyclic dependency tests?

Reply:
I think you might want a test that checks that, at least, specifying 2lh does something (as opposed to just being ignored by the browser instead of correctly cycle-resolved to 2x the default value). Checking that the height becomes taller than the initial one might be enough (or maybe larger than 1.75x more and smaller than 2.25x less?)

Reply:
Actually, that was a good idea, sorry for pushing back initially. Also, while fixing that, I found other problems with the test, and fixed them as well. Can you have a look again?